trunk/NDHMS/Routing/Reservoirs/Level_Pool/module_levelpool.F

```diff
@@ -14,12 +14,13 @@
 ! processes the given inputs, parameters, and
 ! state for a particular level pool reservoir and
 ! returns the output/outflow.
+
 module module_levelpool
 
     use module_levelpool_parameters
     use module_levelpool_state
+    use module_reservoir_utilities
```

Comment:
Best practice is to specify `only` and exactly what is being used. I would stongly prefer this. I know that it was not enforced above, but this requirement should be added to our style guide.

Reply:
Agreed. Though I have logging capabilities in this module that can be uncommented during dev/debugging. Those call logging functions in the utilities module.

Reply:
We could add the logging to they HYDRO_D macro and wrap up those in pre-processing, or maybe we want a logging macro (maybe even a logging macro per module?).  The reservoir logging can be quite expensive.  It is essentially a poor mans parallel debugging utility logging internal module state every time step.  But it can also be quite valuable.

Reply:
YOu can still employ `only` for commented or HYDROD_D pieces of code.
HYDRO_D is a hack. Generally we should move away from all CPP, including compile-time options and make them run time options. 
There should be something akin to a verbosity level dictionary/struct that is passed at run time. 

... in a perfect world. But you go to war with the army you have...

Reply:
@jdmattern-noaa fixing. Creating RESERVOIR_D environment variable for enhanced reservoir logging and importing reservoir utilities to level pool only when RESERVOIR_D is set.

Reply:
I would strongly advise against doing this.

Reply:
@jdmattern-noaa fixing per our discussion. Keeping RESERVOIR_D only in the reservoir module and not adding to any other build or runtime options.